src/server/wesnothd/server.cpp

```diff
@@ -1807,7 +1807,17 @@ void server::handle_player_in_game(socket_ptr socket, simple_wml::document& data
 	} else if(data.child("stop_updates")) {
 		g.send_data(data, socket);
 		return;
-		// Data to ignore.
+	} else if(data.child("game_history_request")) {
+		if(user_handler_) {
+			int offset = data.child("game_history_request")->attr("offset").to_int();
+			std::string player_name = data.child("game_history_request")->attr("search_for").to_string();
+			int player_id = user_handler_->get_forum_id(player_name);
```

Comment:
This should get the ID from the player object if no `search_for` was given. And probably also if search_for references an online player.

Reply:
Is there a case where no `search_for` would be given?

Reply:
I assume `search_for` is given if you want the history for another player?

There should also be an early abort when `get_forum_id` returns 0.

Reply:
Searching for yourself would also make sense, I think.  Or the client could just directly provide the ID since it knows its own forum ID at this point.

Reply:
Searching for yourself would be the primary usecase (at least initially). Since the forum ID is now known for everyone client-side it would be good to support passing that instead of (or in addition) to username.

Reply:
So essentially:
* if the forum ID attribute is present -> use that
* else if the player being searched for is online -> get the forum ID from wesnothd's player info
* else get the forum ID by querying the database, making sure it didn't return a zero before actually running the query

Reply:
Sure. Or we could just make the ID attribute mandatory. Either works.

Reply:
I would not give either `forum_id` or `search_for` when asking for own game history.

Is there even a need for both `forum_id` and `search_for`? I would just pick one. Probably `search_for` so you can search for users you just know the name of. Which is basically what Vultraz suggested initially just with name instead of forum ID.

Reply:
So in that case:
* if `search_for` is not present -> get forum ID from the requesting player's info
* else if `search_for` is present -> search the players currently online and if found get the forum ID from their player info
* else -> query the forum database for their forum ID

Reply:
Why the restriction of needed to be online, though?

Reply:
The last case is for offline players.

Reply:
Alright, updated.